### ISSUE TITLE
updateQueries after mutation is removing the data for a component if data not previously cached.

### ISSUE THREAD

**Issue by jasonmorita** [apollographql/apollo-client#1282]
I am working on a feature with a list of users in a group. Initial query looks like:
```
const query = gql`
    query userGroup($groupId: String!) {
      group(id: $groupId) {
        items {
          id
          status
          participant {
            items {
              email
              id
              title
            }
          }
        }
      }
    }
`;
```

And an `updateQueries` that looks like:
```
                userGroup: (prev, { mutationResult, queryVariables }) => {
                    // some checking here...

                    const updated = update(prev, {
                        group: {
                            items: {
                                0: {
                                    participant: {
                                        items: {
                                            $unshift: [user]
                                        }
                                    }
                                }
                            }
                        }
                    });

                    return updated;
                },
```

If the component initially has the full list of users in it's data payload, I can add and remove users from the list and everything updates fine.

If I load up the component with a single user, then attempt to add a user (the mutation payload doesn't contain the data I need to update the store, but I have access to it in the mutation function closure as `user`), the `data` in the component doesn't contain `group` anymore, just the methods that Apollo client adds to `props.data` like `loading`, etc.

The pattern I am observing is if the user objects I want to add to the items array are NOT present originally, the entire object is not sent back in the `props.data`. If the items are in the cache, then I remove them, then add them back, it all works.

I can inspect the Redux store and see that the items I expect to be there are in the store, but somehow the mapping to the component props is removed.

All the above code examples work, but when the component re-renders, the `props.data` has no data from the query.

**Comment by helfer** [comment]
@jasonmorita Thanks for filing an issue! From your description it's not really clear to me what's going on. If you could make a minimal reproduction, that would be great!

**Comment by jasonmorita** [comment]
I am refactoring this code. Will come back if issue returns.

**Comment by cusxio** [comment]
@jasonmorita I was wondering if you managed to find out the reason why this is happening?

i.e. Why is Apollo removing the query results from the `data` prop on mutation?

**Comment by jasonmorita** [comment]
@cusxio Actually no, because I moved the refetch logic into an async action creator because it seemed like a better way to do it. Unfortunately, since I re-architected that part, I didn't have time to debug it :)

**Comment by xdamman** [comment]
I think I ran into a similar issue.

Whenever I was fetching the list of members of a collective for the `editCollective` page, I was fetching a lot of data: https://github.com/opencollective/frontend/blob/f2b2bc9cd52bea57d15f692621607997dcae1e0c/src/graphql/queries.js#L231-L254

but whenever I'd edit the collective to add a new user to that list, the next render would "forget" about `this.props.data.Collective`.

The problem was that my mutation was returning a subset of that data per member: https://github.com/opencollective/frontend/blob/f2b2bc9cd52bea57d15f692621607997dcae1e0c/src/graphql/mutations.js#L90-L95

I fixed this issue by making sure that I was returning the exact same structure after the mutation.
